src/gtk/spinbutt.cpp

```diff
@@ -186,6 +186,16 @@ void wxSpinButton::SetRange(int minVal, int maxVal)
     GtkDisableEvents();
     gtk_spin_button_set_range((GtkSpinButton*)m_widget, minVal, maxVal);
     m_pos = int(gtk_spin_button_get_value((GtkSpinButton*)m_widget));
+
+    // Gradually reduce the page increment in the case of a narrow range for convenience
+    // and to limit possible up/down ambiguity in gtk_value_changed() when wrapping is on
+    // (The maximal page increment of 10 is consistent with the default page increment set
+    // by calling gtk_spin_button_new_with_range(0, 100, 1) in wxSpinButton::Create().)
+    const int rng = maxVal - minVal;
+    const int inc = 10 - ( (rng < 50) ? 5 : 0 ) - ( (rng < 20) ? 3 : 0 ) - ( (rng < 10) ? 1 : 0 );
```

Comment:
Thanks, I don't disagree with the idea (although I'm not sure if people are not going to be surprised by this), but IMO this would be much more clear if it were written as a sequence of `if`s or, if you really want to keep it in the same expression as,
```suggestion
    const int inc = range < 10 ? 1 : range < 20 ? 2 : range < 50 ? 5 : 10;
```
(I've also renamed the variable to avoid any impression that the increment is determined using the Random Number Generator).

Reply:
OK, I have rewritten it using a set of `if` statements.